tests/foreman/cli/test_bootstrap_script.py

```diff
@@ -15,62 +15,205 @@
+            query={'search': 'name~{0}'.format(prod.name)}
+        )
+
+        cls.ak.add_subscriptions(data={'subscriptions': [{'id': subs[0].id}]})
+
+        # Search for SmartProxy
+        proxy = entities.SmartProxy().search(
+            query={
+                u'search': u'name={0}'.format(
+                    settings.server.hostname)
+                }
+            )
+
+        # Get the arch ID
+        architecture = entities.Architecture().search(
+            query={
+                u'search': u'name="{0}"'.format(DEFAULT_ARCHITECTURE)
+                   }
+        )[0].read()
+
+        # Create the OS
+        os = entities.OperatingSystem().search(query={
+            u'search': u'name="RedHat" AND (major="{0}")'
+            .format(RHEL_7_MAJOR_VERSION)
+            })[0].read()
+
+        cls.domain = entities.Domain(
+            location=[cls.loc],
+            organization=[cls.org],
+                ).create()
+
+        # Create a host group
+        cls.hostgroup = entities.HostGroup(
+            architecture=architecture,
+            content_source=proxy[0].id,
+            domain=[cls.domain.id],
+            location=[cls.loc],
+            content_view=cv,
+            lifecycle_environment=lce.id,
+            organization=[cls.org],
+            operatingsystem=os,
+                ).create()
+
+    @tier4
     def test_positive_register(self):
-        """System is registered
+        """System is registered.
 
         :id: e34561fd-e0d6-4587-84eb-f86bd131aab1
 
         :Steps:
 
-            1. assure system is not registered
-            2. register a system
+            1. create a container with host name
+            2. register system using bootstrap.py
+            3. assert system has been registered
 
         :expectedresults: system is registered, host is created
 
-        :CaseAutomation: notautomated
+        :CaseAutomation: automated
 
-        :CaseImportance: Critical
+        :CaseImportance: High
         """
+        my_host = Container(agent=True)
+        host_name = my_host.execute("hostname")
+        my_fqdn = host_name.strip() + '.' + self.domain.name + '.' 'com'
+        my_host.execute("curl -O http://{}/pub/bootstrap.py".format(settings.server.hostname))
+        result = my_host.execute("python bootstrap.py -l admin -p changeme -s {} -o '{}' "
+                                 "-L '{}' -g {} -a {} --fqdn {} --force --add-domain"
+                                 .format(
+                                     settings.server.hostname,
+                                     self.org.name, self.loc.name,
+                                     self.hostgroup.name,
+                                     self.ak.name,
+                                     my_fqdn
+                                         )
+                                 )
+        assert "The system has been registered" in result, 'Not registered'
+        Container.delete(my_host)
 
     @tier1
```

Comment:
Tier1? This looks comparable to this other test here, which is Tier4.

Reply:
That comes from the original spec, I mean how it was when stubbed out. Tier1 is supposed to be for "Critical component level functional tests". Lets make them both Tier3 as they are quite quick and Tier4 is for test with longer set-up times.

Thank you